### PR TITLE
feat: add 3D OSM building tiles

### PR DESCRIPTION
**Pull request by nina992**
# Overview
OSM Buildings is a 3D buildings layer covering the entire world. It's available as a 3D Tileset on Cesium.
we modified reearth  in way that allows the user to add OSM building 3d Tileset as a layer.
## Memo
backend pr : https://github.com/reearth/reearth/pull/340

### CODE REVIEW

File: src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx [hunk 1]
```diff
@@ -41,11 +43,18 @@ export default function Tileset({ layer }: PrimitiveProps<Property>): JSX.Elemen
...
-
-  return !isVisible || !tileset ? null : (
+  const tilesetUrl = useMemo(() => {
+    return sourceType === "osm" && isVisible
+      ? IonResource.fromAssetId(96188)
+      : sourceType === "url" && isVisible && tileset
+      ? tileset
+      : null;
+  }, [isVisible, sourceType, tileset]);
+  console.log(tilesetUrl);
```
Comment (KaWaite) [inline src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx:53]:
```suggestion
```

File: src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx [hunk 2]
```diff
@@ -41,11 +43,18 @@ export default function Tileset({ layer }: PrimitiveProps<Property>): JSX.Elemen
       setStyle(new Cesium3DTileStyle(await res.json()));
     })();
   }, [styleUrl]);
-
-  return !isVisible || !tileset ? null : (
+  const tilesetUrl = useMemo(() => {
```
Comment (KaWaite) [inline src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx:46]:
```suggestion

  const tilesetUrl = useMemo(() => {
```

File: src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx [hunk 3]
```diff
@@ -41,11 +43,18 @@ export default function Tileset({ layer }: PrimitiveProps<Property>): JSX.Elemen
...
+    return sourceType === "osm" && isVisible
+      ? IonResource.fromAssetId(96188)
```
Comment (KaWaite) [inline src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx:48]: Please leave a comment here at least mentioning that this is will retrieve the OSM URL and/or a link to the code on github that you are referencing for this. Makes it easier for other devs to know what this is about.

File: src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx [hunk 4]
```diff
@@ -42,10 +44,18 @@ export default function Tileset({ layer }: PrimitiveProps<Property>): JSX.Elemen
     })();
   }, [styleUrl]);
 
-  return !isVisible || !tileset ? null : (
+  const tilesetUrl = useMemo(() => {
+    return sourceType === "osm" && isVisible
+      ? //https://github.com/CesiumGS/cesium/blob/1.69/Source/Scene/createOsmBuildings.js#L50
+        IonResource.fromAssetId(96188)
```
Comment (KaWaite) [inline src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx:50]:
```suggestion
      ?
        IonResource.fromAssetId(96188) //https://github.com/CesiumGS/cesium/blob/1.69/Source/Scene/createOsmBuildings.js#L50
```

File: src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx [hunk 5]
```diff
@@ -42,10 +44,18 @@ export default function Tileset({ layer }: PrimitiveProps<Property>): JSX.Elemen
...
+      : sourceType === "url" && isVisible && tileset
+      ? tileset
+      : null;
+  }, [isVisible, sourceType, tileset]);
+  return !isVisible || (!tileset && !sourceType) || !tilesetUrl ? null : (
```
Comment (KaWaite) [inline src/components/molecules/Visualizer/Engine/Cesium/Tileset/index.tsx:55]:
```suggestion
  }, [isVisible, sourceType, tileset]);
  
  return !isVisible || (!tileset && !sourceType) || !tilesetUrl ? null : (
```

**Review by KaWaite — APPROVED** [review]

**Review by rot1024 — APPROVED** [review]

**Pull request merged**